### PR TITLE
test(gnovm): test all files in gnovm/tests subdirs

### PR DESCRIPTION
**Pull request by thehowl**
Changes the EvalFiles tests to support sub-directories in `gnovm/tests`, skipping the `extern` directory as it is used for imports.

Other changes mostly concern not having tests where some output lines terminate with spaces:

- Changed FuncValue.String, so that in case of a closure value the signature is printed, as shown in the `assign_unnamed_type` tests.
- Require all directives to be on a new line (mostly for consistency with 99% of the testing corpus)
- use commas in some places instead of spaces; use `;` for end of line on `Println` statements when they're supposed to print an empty string.
- remove the leading space in the `what the firetruck?` tests.

<details><summary>Contributors' checklist...</summary>

- [x] Added new tests, or not needed, or not feasible
- [x] Provided an example (e.g. screenshot) to aid review or the PR is self-explanatory
- [x] Updated the official documentation or not needed
- [x] No breaking changes were made, or a `BREAKING CHANGE: xxx` message was included in the description
- [x] Added references to related issues and PRs
- [x] Provided any useful hints for running manual tests
- [x] Added new benchmarks to [generated graphs](https://gnoland.github.io/benchmarks), if any. More info [here](https://github.com/gnolang/gno/blob/master/.benchmarks/README.md).
</details>

### CODE REVIEW

File: gnovm/pkg/gnolang/eval_test.go [hunk 1]
```diff
@@ -88,7 +106,10 @@ func commentFrom(s string, delims []string) []string {
 			next = directivesFound[i+1].index
 		}
 
-		directivesFound[i].res = strings.TrimSpace(strings.ReplaceAll(s[directivesFound[i].index+len(directivesFound[i].delim):next], "\n// ", "\n"))
+		parsed := reCommentPrefix.ReplaceAllLiteralString(
```
Comment (deelawn) [inline gnovm/pkg/gnolang/eval_test.go:109]: I know this was a long line of code before you change, but a comment here to say what this does would go a long way for readability.

**Review by deelawn — APPROVED** [review]
Nice work. So until now, any tests in subdirectories weren't being executed? Good catch. I made a couple suggestions.

File: gnovm/pkg/gnolang/eval_test.go (hunk 1 above)
Comment (thehowl) [reply gnovm/pkg/gnolang/eval_test.go:109]: Split up a bit and added some comments :heavy_check_mark:

**Comment by thehowl** [comment]
> Nice work. So until now, any tests in subdirectories weren't being executed? Good catch. I made a couple suggestions.

They're only tested in gnovm/tests. Alas, the tests in gnovm/tests/files are checked twice. Ideally, I'd remove all Go code in gnovm/tests, so we can:

1. remove filetests for users -- #1346.
2. have top-level `tests/` directory be used as central points for integration tests.
3. move code for the inner workings of `gno test` to `cmd/gno` or some other package in `pkg`.

But it's going to be a while before we get there.

**Pull request merged**